include/kokkos-utils/view/slice.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef KOKKOS_UTILS_VIEW_SLICE_HPP
+#define KOKKOS_UTILS_VIEW_SLICE_HPP
+
+#include "kokkos-utils/concepts/View.hpp"
+
+/**
+ * @file
+ *
+ * This file is a collection of helper functions similar to what can be found in
+ * the standard @c mdspan header, related to retrieving slices of a @c Kokkos::View.
+ *
+ * References:
+ *  - https://en.cppreference.com/w/cpp/container/mdspan
+ *  - https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2630r4.html
+ */
+
+namespace Kokkos::utils::view
+{
+
+namespace impl
+{
+
+template <typename ViewType, typename AllsValues, size_t... AllsIndices, typename... Indices>
+KOKKOS_FUNCTION
+constexpr auto slice(ViewType&& view, AllsValues&& alls, std::index_sequence<AllsIndices...>, Indices&&... indices) {
```

Comment:
Just for the naming :) We could also rename `Indices&&... indices` as `SliceArgs&&... args` to reuse the names from the submdspan paper. It would also help see that it's not necessarily integers, but it can also be pairs, alls, ...

Reply:
After playing a bit with the code, I also have two variants to propose. The first variant would be to create the `alls` on the fly as needed, so it would avoid passing it as a parameter. We would have:

```
template <typename ViewType, typename... Indices, size_t... AllsIndices>
KOKKOS_FUNCTION
constexpr auto slice(ViewType&& view, Indices&&... indices, std::index_sequence<AllsIndices...>) {    
    return Kokkos::subview(
        std::forward<ViewType>(view),
        std::forward<Indices>(indices)...,
        std::get<AllsIndices>(std::array<Kokkos::ALL_t, sizeof...(AllsIndices)>{})...
    );
}

template <size_t Rank, typename ViewType, typename... Indices>
requires concepts::ViewOfRank<std::remove_reference_t<ViewType>, Rank>
KOKKOS_FUNCTION
constexpr auto slice(ViewType&& view, Indices&&... indices)
{
    constexpr auto size = Rank - sizeof...(Indices);
    return impl::slice<ViewType, Indices...>(
        std::forward<ViewType>(view),
        std::forward<Indices>(indices)...,
        std::make_index_sequence<size>{}
    );
}
```

The second variant would replace the helper function in the `impl` namespace with a lambda. So the advantage would be that the code would be shorter. We'd have:

```
template <size_t Rank, typename ViewType, typename... Indices>
requires concepts::ViewOfRank<std::remove_reference_t<ViewType>, Rank>
KOKKOS_FUNCTION
constexpr auto slice(ViewType&& view, Indices&&... indices)
{
    constexpr auto size = Rank - sizeof...(Indices);
    return [view = std::forward<ViewType>(view), ... indices = std::forward<Indices>(indices)] 
        <size_t... AllsIndices>(std::index_sequence<AllsIndices...>) {
            return Kokkos::subview(
                view,
                indices...,
                std::get<AllsIndices>(std::array<Kokkos::ALL_t, size>{})...
            );
        }(std::make_index_sequence<size>{});
}
```

Up to you. We can keep the current version, or one of these variants.

Reply:
Done.